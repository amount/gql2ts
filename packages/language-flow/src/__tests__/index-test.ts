import {
  FLOW_ENUM_TYPE,
  FLOW_FORMAT_ENUM,
  FLOW_POST_PROCESSOR,
  FLOW_WRAP_PARTIAL
} from '../';
import { GraphQLEnumValue } from 'graphql';

describe('language-flow', () => {
  describe('FLOW_FORMAT_ENUM', () => {
    it('w/o description', () => {
      expect(
        FLOW_FORMAT_ENUM(
          [
            { name: 'a' } as GraphQLEnumValue,
            { name: 'b' } as GraphQLEnumValue
          ],
          () => ''
        )
      ).toMatchSnapshot();
    });
    it('w/ description', () => {
      expect(
        FLOW_FORMAT_ENUM(
          [
            { name: 'a', description: 'value A' } as GraphQLEnumValue,
            { name: 'b' } as GraphQLEnumValue
          ],
          () => ''
        )
      ).toMatchSnapshot();
    });
    it('w/ deprecated value', () => {
      expect(
        FLOW_FORMAT_ENUM(
          [
            {
              name: 'a',
              description: 'value A',
              deprecationReason: 'Bad',
              isDeprecated: true
            } as GraphQLEnumValue,
            {
              name: 'b',
              deprecationReason: 'Bad',
              isDeprecated: true
            } as GraphQLEnumValue,
            { name: 'c', description: 'value C' } as GraphQLEnumValue,
            { name: 'd' } as GraphQLEnumValue,
            { name: 'e', isDeprecated: false } as GraphQLEnumValue
          ],
          () => ''
        )
      ).toMatchSnapshot();
    });
  });

  describe('FLOW_ENUM_TYPE', () => {
    it('formats properly', () => {
      expect(
        FLOW_ENUM_TYPE(
          'Test',
          FLOW_FORMAT_ENUM(
            [
              { name: 'a' } as GraphQLEnumValue,
              { name: 'b' } as GraphQLEnumValue
            ],
            () => ''
          )
        )
      ).toMatchSnapshot();
    });
  });
});
