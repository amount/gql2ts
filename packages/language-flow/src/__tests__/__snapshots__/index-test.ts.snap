// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`language-flow FLOW_ENUM_TYPE formats properly 1`] = `"type Test = 'a' | 'b'"`;

exports[`language-flow FLOW_FORMAT_ENUM w/ deprecated value 1`] = `"'a' | 'b' | 'c' | 'd' | 'e'"`;

exports[`language-flow FLOW_FORMAT_ENUM w/ description 1`] = `"'a' | 'b'"`;

exports[`language-flow FLOW_FORMAT_ENUM w/o description 1`] = `"'a' | 'b'"`;
